blockchain/state/statedb.go

```diff
@@ -541,34 +572,62 @@ func (self *StateDB) deleteStateObject(stateObject *stateObject) {
 	self.setError(self.trie.TryDelete(addr[:]))
 }
 
-// Retrieve a state object given by the address. Returns nil if not found.
+// getStateObject retrieves a state object given by the address, returning nil if
+// the object is not found or was deleted in this execution context. If you need
+// to differentiate between non-existent/just-deleted, use getDeletedStateObject.
 func (self *StateDB) getStateObject(addr common.Address) *stateObject {
+	if obj := self.getDeletedStateObject(addr); obj != nil && !obj.deleted {
+		return obj
+	}
+	return nil
+}
```

Comment:
Is `obj.deleted` atomic value or if statement is thread-safe? 
Then we can surely believe about object state in execution context.

Reply:
Hmm.. it is not, but is it necessary? Do you have any concern about this?

Reply:
I think it's ok because the DB guarantees concurrency.